### ISSUE TITLE
Add Python bindings

### ISSUE THREAD

**Issue by arobsn** [ergoplatform/sigma-rust#780]
Implement an API that is functionally equivalent to the Wasm bindings (`ergo-lib-wasm`). Essentially, this API should provide all the necessary functionality for dApps to build and sign transactions. Key features include:

- Transaction building
- Input selection
- Transaction signing
- Sigma Constants management and [de]serialization

**Comment by marctheshark3** [comment]
If I wanted to try this out, is it correct to fork the repo build this out, and then submit PR?

**Comment by sethdusek** [comment]
> If I wanted to try this out, is it correct to fork the repo build this out, and then submit PR?

Yes, check out the bindings folder to see how other bindings are implemented, and the pyo3 crate if you're interested: https://docs.rs/crate/pyo3/latest

**Comment by 0xf965** [comment]
Is this task already taken? @SethDusek  are you currently working on it? If not, I can do it.

**Comment by sethdusek** [comment]
> Is this task already taken? [@SethDusek](https://github.com/SethDusek) are you currently working on it? If not, I can do it.

Hi, I'll try to allocate more time to this issue soon. If I'm unable to I'll let you know. Thanks!

**Comment by sethdusek** [comment]
> Implement an API that is functionally equivalent to the Wasm bindings (`ergo-lib-wasm`). Essentially, this API should provide all the necessary functionality for dApps to build and sign transactions. Key features include:
> 
>     * Transaction building
> 
>     * Input selection
> 
>     * Transaction signing
> 
>     * Sigma Constants management and [de]serialization

I've published documentation and an alpha version to pypi to try out (will be yanked before final release). I would appreciate any feedback on the interface https://ergo-lib-python.readthedocs.io
